asset-transfer-ledger-queries/chaincode-go/asset_transfer_ledger_chaincode.go

```diff
@@ -393,17 +395,26 @@ func (t *SimpleChaincode) GetAssetHistory(ctx contractapi.TransactionContextInte
 		}
 
 		var asset Asset
-		err = json.Unmarshal(response.Value, &asset)
-		if err != nil {
-			return nil, err
+		if len(response.Value) > 0 {
+			err = json.Unmarshal(response.Value, &asset)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			asset = Asset{
+				ID: assetID,
+			}
 		}
 
 		timestamp, err := ptypes.Timestamp(response.Timestamp)
 		if err != nil {
 			return nil, err
 		}
+
+		log.Printf("GetAssetHistory: response.IsDelete %v", response.IsDelete)
```

Comment:
I don't mind having a single log statement at the beginning (as above) and end of each function, but I assume this was just added for your debugging.

Reply:
done